js/src/common/Store.ts

```diff
@@ -0,0 +1,243 @@
+import app from '../common/app';
+import { FlarumRequestOptions } from './Application';
+import Model, { ModelData, SavedModelData } from './Model';
+
+export interface ApiQueryParamsSingle {
+  fields?: string[];
+  include?: string;
+  bySlug?: boolean;
+  meta?: {
+    [key: string]: any;
+  };
+}
+
+export interface ApiQueryParamsPlural {
+  fields?: string[];
+  include?: string;
+  filter?: {
+    q: string;
+    [key: string]: string;
+  };
+  page?: {
+    offset?: number;
+    number?: number;
+    limit?: number;
+    size?: number;
+  };
+  sort?: string;
+  meta?: {
+    [key: string]: any;
+  };
+}
+
+export type ApiQueryParams = ApiQueryParamsPlural | ApiQueryParamsSingle;
+
+export interface ApiPayloadSingle {
+  data: SavedModelData;
+  included?: SavedModelData[];
+}
+
+export interface ApiPayloadPlural {
+  data: SavedModelData[];
+  included?: SavedModelData[];
+  links?: {
+    first: string;
+    next?: string;
+    prev?: string;
+  };
+}
```

Comment:
The meta I mentioned before was about the response payload here btw, not the request, didn't even know the request had that in the spec.